### PR TITLE
Always use computer domain to bind to RootDSE

### PR DESCRIPTION
**Pull request by bill-long**
**Issue:**
If a user is logged in to a machine in the Exchange forest using a user account from a trusted forest, binding to LDAP://RootDSE gives us the user forest.

**Fix:**
Get the computer's domain and explicitly bind to that RootDSE.

### CODE REVIEW

File: Setup/SetupAssist/Checks/Domain/Test-ExchangeADSetupLevel.ps1 [hunk 1]
```diff
@@ -134,7 +134,7 @@ function Test-ExchangeADSetupLevel {
     }
 
     function GetExchangeADSetupLevel {
-        $rootDSE = [ADSI]("LDAP://RootDSE")
+        $rootDSE = [ADSI]("LDAP://$([System.DirectoryServices.ActiveDirectory.Domain]::GetComputerDomain().Name)/RootDSE")
```
Comment (dpaulson45) [inline Setup/SetupAssist/Checks/Domain/Test-ExchangeADSetupLevel.ps1:137]: This is the same on line 148, could just set `$rootDSE` once then in this function.

**Comment by dpaulson45** [comment]
Should probably do the same thing for `$rootDSE = [ADSI]("GC://RootDSE")` as well.

File: Setup/SetupAssist/Checks/Domain/Test-ExchangeADSetupLevel.ps1 (hunk 1 above)
Comment (bill-long) [reply Setup/SetupAssist/Checks/Domain/Test-ExchangeADSetupLevel.ps1:137]: Good point. Not sure why that's in there twice.

**Comment by lusassl-msft** [comment]
I'm wondering if we need to adjust the `Test-CVE-2021-34470.ps1` function as well. 

```
$schemaMaster = [System.DirectoryServices.ActiveDirectory.Forest]::GetCurrentForest().SchemaRoleOwner
```

GetCurrentForest() --> Gets a Forest object for the current user context.
https://learn.microsoft.com/en-us/dotnet/api/system.directoryservices.activedirectory.forest.getcurrentforest?view=netframework-4.7.2#system-directoryservices-activedirectory-forest-getcurrentforest

**Comment by bill-long** [comment]
Good point. In fact, all calls to GetCurrentForest() need to be changed to GetComputerDomain().Forest.

**Review by lusassl-msft — APPROVED** [review]
Found no other functions that use the problematic code. Tested HealthChecker and it seems to work fine with the updated calls (as the return object should be the same). Approved and ready to merge from my point of view.

**Pull request merged**